### PR TITLE
feat: preview support

### PR DESCRIPTION
**Pull request by yaegassy**
Currently it just jumps to the target location in the editor, but I made a simplified version for now.

**DEMO(mp4)**:

https://user-images.githubusercontent.com/188642/165060816-724ddac1-7eea-459f-9c28-127d1ed34084.mp4

### CODE REVIEW

**Comment by johnsoncodehk** [comment]
Look great! 🤩

Do you know coc has any plan to adding selection range api?

**Comment by yaegassy** [comment]
@johnsoncodehk 

> Do you know coc has any plan to adding selection range api?

Not sure if there is a plan. Vim can handle this in a different way, so I adjusted the behavior like the selection api

**Comment by yaegassy** [comment]
**preview (selection ver)**:

https://user-images.githubusercontent.com/188642/165156070-268dae57-75f0-4ebd-a133-eed520cfd8fc.mp4

**Comment by yaegassy** [comment]
@johnsoncodehk "Nuxt" preview does not work correctly in coc-volar-tools. Do you know the cause?

### Error

```
[nuxt] [request error] defineNuxtPlugin is not defined
  at $id_f811a61b (./.nuxt/dist/server/server.mjs:2850:1)
  at __instantiateModule__ (./.nuxt/dist/server/server.mjs:4143:9)
  at __ssrLoadModule__ (./.nuxt/dist/server/server.mjs:4081:25)
  at ssrImport (./.nuxt/dist/server/server.mjs:4106:13)
  at $id_5547b628 (./.nuxt/dist/server/server.mjs:1570:37)
  at async __instantiateModule__ (./.nuxt/dist/server/server.mjs:4143:3)
```

### DEMO (mp4)

https://user-images.githubusercontent.com/188642/165193484-033c2d45-91ef-445c-8ea0-a44d780dc787.mp4

### Misc

As a supplement, the actual command looks like this.

**coc-volar-tools**:

> preview/bin/nuxi.js

Maybe `@volar/preview/bin/nuxi/plugin.ts` is related to something.

```
node "/Users/yaegassy/sandbox/coc-volar-tools/node_modules/@volar/preview/bin/nuxi.js" dev --port 3333
```

**VSCode**:

> dist/preview-bin/nuxi.js

```
node "/Users/yaegassy/sandbox/volar/extensions/vscode-vue-language-features/dist/preview-bin/nuxi.js" dev --port 3333
```

**Comment by johnsoncodehk** [comment]
Can it works if you run below commands manually?
```
$ cd /Users/yaegassy/sandbox/npm-nuxt-app
$ node "/Users/yaegassy/sandbox/volar/extensions/vscode-vue-language-features/dist/preview-bin/nuxi.js" dev --port 3333
```

**Comment by yaegassy** [comment]
Yes, using nuxi.js of the volar extensions side, the server starts up correctly.

**Comment by johnsoncodehk** [comment]
Can this problem fix if you copy `/Users/yaegassy/sandbox/volar/extensions/vscode-vue-language-features/dist/preview-bin` files to `/Users/yaegassy/sandbox/coc-volar-tools/node_modules/@volar/preview/bin` and run below commands?
```
$ cd /Users/yaegassy/sandbox/npm-nuxt-app
$ node "/Users/yaegassy/sandbox/coc-volar-tools/node_modules/@volar/preview/bin/nuxi.js" dev --port 3333
```

If it fix, what are these files different?

**Comment by yaegassy** [comment]
This procedure can also be reproduced.

```
mkdir -p /tmp/volar-preview
cd /tmp/volar-preview
npm init -y
npm i @volar/preview

# move nuxt3 project
cd /Users/yaegassy/sandbox/npm-nuxt-app
node "/tmp/volar-preview/node_modules/@volar/preview/bin/nuxi.js" dev --port 3333
```

Then access `http://localhost:3333` in your browser to reproduce

**Comment by yaegassy** [comment]
There seems to be no difference in the `nuxi.js` file itself...

```
diff /Users/yaegassy/sandbox/volar/extensions/vscode-vue-language-features/dist/preview-bin/nuxi.js /tmp/volar-preview/node_modules/@volar/preview/bin/nuxi.js
```

The result is empty, so there is no difference...

**Comment by johnsoncodehk** [comment]
@yaegassy Please try `@volar/preview@0.34.10-patch.1`.

**Comment by yaegassy** [comment]
> Please try @volar/preview@0.34.10-patch.1.

Thanks for the adjustment. It works! 🙇

File: README.md [hunk 1]
```diff
@@ -23,10 +23,13 @@ Plug 'yaegassy/coc-volar-tools', {'do': 'yarn install --frozen-lockfile'}
 ## Configuration options
 
 - `volar-tools.enable`: Enable coc-volar-tools extension, default: `true`
+- `volar.preview.port`: Default port for component preview server, default: `3333`
```
Comment (johnsoncodehk) [inline README.md]: It should be:
```diff
- `volar.preview.port`: Default port for component preview server, default: `3333`
+ `volar.preview.port`: Default port for preview server, default: `3333`
```

Comment (yaegassy) [reply README.md]: Thanks.

**Pull request merged**